baldrick/plugins/github_pull_requests.py

```diff
@@ -148,13 +152,16 @@ def process_pull_request(repository, number, installation, is_new=False):
         if full_context.startswith(current_app.bot_username + ':'):
```

Comment:
This is iterating over `existing_statuses` but setting checks, that seems like it's not going to work correctly.

Reply:
Oh yeah, good point. But does it make sense to iterate over `existing_checks`? Hmm... Gotta find time to dig at the API doc.